pkg/cmd/serve/serve.go

```diff
@@ -22,94 +22,109 @@ import (
 	"github.com/authzed/spicedb/pkg/validationfile"
 )
 
-func registerServeCmd(rootCmd *cobra.Command) {
-	var datastoreOptions cmdlib.DatastoreConfig
-	serveCmd := &cobra.Command{
-		Use:     "serve",
-		Short:   "serve the permissions database",
-		Long:    "A database that stores, computes, and validates application permissions",
-		PreRunE: defaultPreRunE,
-		Run: func(cmd *cobra.Command, args []string) {
-			serveRun(cmd, args, datastoreOptions)
-		},
-		Example: fmt.Sprintf(`	%s:
-		spicedb serve --grpc-preshared-key "somerandomkeyhere"
-
-	%s:
-		spicedb serve --grpc-preshared-key "realkeyhere" --grpc-tls-cert-path path/to/tls/cert --grpc-tls-key-path path/to/tls/key \
-			--http-tls-cert-path path/to/tls/cert --http-tls-key-path path/to/tls/key \
-			--datastore-engine postgres --datastore-conn-uri "postgres-connection-string-here"
-`, color.YellowString("No TLS and in-memory"), color.GreenString("TLS and a real datastore")),
-	}
-
+func RegisterServeFlags(cmd *cobra.Command) {
 	// Flags for the gRPC API server
-	cobrautil.RegisterGrpcServerFlags(serveCmd.Flags(), "grpc", "gRPC", ":50051", true)
-	serveCmd.Flags().String("grpc-preshared-key", "", "preshared key to require for authenticated requests")
-	serveCmd.Flags().Duration("grpc-shutdown-grace-period", 0*time.Second, "amount of time after receiving sigint to continue serving")
-	if err := serveCmd.MarkFlagRequired("grpc-preshared-key"); err != nil {
+	cobrautil.RegisterGrpcServerFlags(cmd.Flags(), "grpc", "gRPC", ":50051", true)
+	cmd.Flags().String("grpc-preshared-key", "", "preshared key to require for authenticated requests")
+	cmd.Flags().Duration("grpc-shutdown-grace-period", 0*time.Second, "amount of time after receiving sigint to continue serving")
+	if err := cmd.MarkFlagRequired("grpc-preshared-key"); err != nil {
 		panic("failed to mark flag as required: " + err.Error())
 	}
 
 	// Flags for the datastore
-	cmdlib.RegisterDatastoreFlags(serveCmd, &datastoreOptions)
-	serveCmd.Flags().Bool("datastore-readonly", false, "set the service to read-only mode")
-	serveCmd.Flags().StringSlice("datastore-bootstrap-files", []string{}, "bootstrap data yaml files to load")
-	serveCmd.Flags().Bool("datastore-bootstrap-overwrite", false, "overwrite any existing data with bootstrap data")
-
-	serveCmd.Flags().Bool("datastore-request-hedging", true, "enable request hedging")
-	serveCmd.Flags().Duration("datastore-request-hedging-initial-slow-value", 10*time.Millisecond, "initial value to use for slow datastore requests, before statistics have been collected")
-	serveCmd.Flags().Uint64("datastore-request-hedging-max-requests", 1_000_000, "maximum number of historical requests to consider")
-	serveCmd.Flags().Float64("datastore-request-hedging-quantile", 0.95, "quantile of historical datastore request time over which a request will be considered slow")
+	cmd.Flags().String("datastore-engine", "memory", `type of datastore to initialize ("memory", "postgres", "cockroachdb")`)
```

Comment:
Shouldn't this use `RegisterDatastoreFlags`?

Reply:
Yep! Missed that in the rebase! thanks